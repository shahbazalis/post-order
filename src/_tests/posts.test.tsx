import React from "react";
import { render, RenderResult } from "@testing-library/react";
import Enzyme from "enzyme";

import Posts from "../views/posts/posts";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
const { createMemoryHistory } = require("history");
import { Router } from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe("<Posts/>", () => {
  it("renders with Props", async () => {
    let history = createMemoryHistory();
    let wrapper: RenderResult;
    wrapper = render(
      <Router location={history.location} navigator={history}>
        <Posts />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
