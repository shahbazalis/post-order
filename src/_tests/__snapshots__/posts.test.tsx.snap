// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> renders with Props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="loading-spinner"
        />
        <div>
          <h1
            class="heading-1"
          >
            Post List
          </h1>
          <button
            class="button"
            data-testid="logout-btn"
          >
            Log Out
          </button>
        </div>
        <div
          class="grid-container"
        >
          <div>
            <div>
              <input
                class="search-input"
                data-testid="search-name"
                name="search"
                placeholder="Search"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <div
              class="fleft"
            >
              <button
                class="ml15"
                data-testid="prv-page-btn"
                disabled=""
              >
                Previous Page
              </button>
            </div>
            <div
              class="fleft"
            >
              <button
                class="ml15"
                data-testid="next-page-btn"
              >
                Next Page
              </button>
            </div>
          </div>
          <div>
            <div
              class="fleft"
            >
              <button
                class="ml15"
                data-testid="sort-recent-msg"
              >
                Recent Posts
              </button>
            </div>
            <div>
              <button
                class="ml15"
                data-testid="sort-old-msg"
              >
                Old Posts
              </button>
            </div>
          </div>
          <div>
            <div>
              <input
                class="search-input"
                data-testid="search-name"
                name="search"
                placeholder="Search"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <ul
            class="first-list"
          />
        </div>
        <div>
          <div>
            <ul
              class="second-list"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="loading-spinner"
      />
      <div>
        <h1
          class="heading-1"
        >
          Post List
        </h1>
        <button
          class="button"
          data-testid="logout-btn"
        >
          Log Out
        </button>
      </div>
      <div
        class="grid-container"
      >
        <div>
          <div>
            <input
              class="search-input"
              data-testid="search-name"
              name="search"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <div
            class="fleft"
          >
            <button
              class="ml15"
              data-testid="prv-page-btn"
              disabled=""
            >
              Previous Page
            </button>
          </div>
          <div
            class="fleft"
          >
            <button
              class="ml15"
              data-testid="next-page-btn"
            >
              Next Page
            </button>
          </div>
        </div>
        <div>
          <div
            class="fleft"
          >
            <button
              class="ml15"
              data-testid="sort-recent-msg"
            >
              Recent Posts
            </button>
          </div>
          <div>
            <button
              class="ml15"
              data-testid="sort-old-msg"
            >
              Old Posts
            </button>
          </div>
        </div>
        <div>
          <div>
            <input
              class="search-input"
              data-testid="search-name"
              name="search"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <ul
          class="first-list"
        />
      </div>
      <div>
        <div>
          <ul
            class="second-list"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
